client/rpc-api/src/state/mod.rs

```diff
@@ -39,10 +39,21 @@ pub trait StateApi<Hash> {
 	#[rpc(name = "state_call", alias("state_callAt"))]
 	fn call(&self, name: String, bytes: Bytes, hash: Option<Hash>) -> FutureResult<Bytes>;
 
-	/// Returns the keys with prefix, leave empty to get all the keys
+	/// DEPRECATED: Please use `state_getKeysPaged` with proper paging support.
+	/// Returns the keys with prefix, leave empty to get all the keys.
 	#[rpc(name = "state_getKeys")]
 	fn storage_keys(&self, prefix: StorageKey, hash: Option<Hash>) -> FutureResult<Vec<StorageKey>>;
 
+	/// Returns the keys with prefix with pagination support. Max count is 10 000.
+	#[rpc(name = "state_getKeysPaged", alias("state_getKeysPagedAt"))]
+	fn storage_keys_paged(
+		&self,
+		prefix: StorageKey,
```

Comment:
If the prefix is optional, I would say that we make it clear by using an `Option`.

Reply:
prefix is not optional. it can be an empty vec which iterates all keys (as they all prefixed with a zero-length key)

Reply:
I'm not sure what you mean by "zero-length key". Such a construct does not exists. The one that is doing the prefix match is you with `starts_with` that returns `true` for the empty slice. Api wise `Option` is much cleaner. That `None` and `Some("")` will return the same values is okay, but `Option` makes the api much better.